ext/libxml/libxml.c

```diff
@@ -784,10 +784,12 @@ static xmlParserInputPtr _php_libxml_external_entity_loader(const char *URL,
 	if (ret == NULL) {
 		if (resource == NULL) {
 			if (ID == NULL) {
-				ID = "NULL";
+				php_libxml_ctx_error(context,
+						"Failed to load external entity because the resolver function returned null\n");
```

Comment:
Wouldn't it make more sense to move this warning where the RETVAL is handled?

Reply:
I don't know. There are quite a few possible paths that lead to this branch. I think refactoring that complicates things more than what we gain? I could be wrong or misunderstanding ofc.

Reply:
I was thinking of just moving the warning for the ``null`` type where the ``// else retval is null`` comment is.

But that might complicate the logic for the rest :/

Reply:
Right I see, but then we'll still need a NULL-check for ID at line 786 so we don't get a double warning (and a NULL deref of ID). So I don't think we gain much by that. I prefer to keep the structure as-is.

Reply:
ACK